packages/mui-material/src/Drawer/Drawer.js

```diff
@@ -239,6 +239,7 @@ const Drawer = React.forwardRef(function Drawer(inProps, ref) {
     <DrawerPaper
       elevation={variant === 'temporary' ? elevation : 0}
       square
+      tabIndex={-1}
```

Comment:
This should also check for `disableInitialContentFocus` through `ModalProps.disableInitialContentFocus`
```suggestion
      tabIndex={ModalProps.disableInitialContentFocus ? undefined : -1}
```